tests/Feature/Posts/FetchPostsTest.php

```diff
@@ -2,78 +2,92 @@
-
-    private $utility;
-
-    protected function setUp(): void
-    {
-        parent::setUp();
-        $this->utility = new Utility($this);
-        $this->utility->testSetup();
-    }
 
     /** @test */
-    public function it_allows_non_users_to_access_posts()
+    public function it_returns_specific_shape()
     {
+        factory(Post::class, 5)->create();
         $this->getJson(route('api.posts'))
-            ->assertOk()
             ->assertJsonStructure([
-                [
-                    'id',
-                    'name',
-                    'content',
-                    'user_id',
-                    'status_id',
-                    'published_at',
-                    'created_at',
-                    'comments' => [
-                        [
-                            'id',
-                            'post_id',
-                            'parent_id',
-                            'user_id',
-                            'text'
+                'data' => [
+                    [
+                        'id',
+                        'name',
+                        'content',
+                        'user_id',
+                        'status_id',
+                        'published_at',
+                        'created_at',
+                        'comments' => [
+                            [
+                                'id',
+                                'post_id',
+                                'parent_id',
+                                'user_id',
+                                'text'
+                            ]
                         ]
                     ]
                 ]
             ]);
     }
 
+    /** @test */
+    public function it_allows_non_users_to_access_posts()
+    {
+        $this->getJson(route('api.posts'))
+            ->assertOk();
+    }
+
     /** @test */
     public function it_allows_users_to_access_posts()
     {
-        $this->utility->loginAdmin();
+        $this->loginAdmin();
         $this->getJson(route('api.posts'))
-            ->assertOk()
-            ->assertJsonStructure([
-                [
-                    'id',
-                    'name',
-                    'content',
-                    'user_id',
-                    'status_id',
-                    'published_at',
-                    'created_at',
-                    'comments' => [
-                        [
-                            'id',
-                            'post_id',
-                            'parent_id',
-                            'user_id',
-                            'text'
-                        ]
-                    ]
-                ]
+            ->assertOk();
+    }
+
+    /** @test */
+    public function it_brings_back_only_published_posts()
+    {
+        $posts = factory(Post::class, 1)->create();
+
+        $response = $this->getJson(route('api.posts'));
+
+        $posts->each(function ($item) use ($response) {
```

Comment:
Since there is only one post here you can just assert this json fragment without the each